packages/salesforcedx-apex-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -211,6 +262,20 @@ export class ApexReplayDebug extends LoggingDebugSession {
     this.printToDebugConsole(
       nls.localize('session_started_text', this.logContext.getLogFileName())
     );
+    // If the projectPath isn't set then don't bother with heap dump processing
+    if (this.projectPath) {
+      if (this.logContext.scanLogForHeapDumpLines()) {
+        if (
+          !await this.logContext.fetchOverlayResultsForApexHeapDumps(
```

Comment:
This is the reason why launchRequest had to be made async. The REST calls are all async and we want to wait on them.

Reply:
Could you collapse these all into one if statement?

Reply:
I collapsed the first two into a single, the last one it still it's own

Reply:
Interesting.